captain/op_gossip.go

```diff
@@ -94,7 +94,7 @@ func (op *GossipOp) sendMsg(msgType GossipMsgType, data []byte) {
 		varint.Pack8(uint8(msgType)),
 		data,
 	)
-	err := op.controller.OpSendWithTimeout(op, c, time.Second)
+	err := op.controller.OpSend(op, c, time.Second, false)
 	if err != nil {
```

Comment:
Instead of passing the timeout and a priority boolean we could use functional options like:
```golang
type Option func(opts OpSendOptions)

func WithTimeout(timeout time.Duration) Option {
    return func(cfg OpSendOptions) {
        cfg.timeout = timeout
    }
}

func WithHighPriority() Option {
    return func(cfg OpSendOptions) {
        cfg.highPrio = true
    }
}

// ....

op.controller.OpSend(op, WithHighPriority(), WithTimout(time.Second))
```

this way it's easier to extend in the future and also keeps more readable as you don't have any "bool" args that always needs a function doc lookup.